platforms/core-runtime/logging/src/main/java/org/gradle/internal/logging/sink/AnsiConsoleUtil.java

```diff
@@ -1,143 +0,0 @@
-/*
- * Copyright 2017 the original author or authors.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *      http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package org.gradle.internal.logging.sink;
-
-import org.apache.commons.io.output.ProxyOutputStream;
-import org.fusesource.jansi.AnsiOutputStream;
-import org.fusesource.jansi.WindowsAnsiPrintStream;
-import org.gradle.internal.os.OperatingSystem;
-
-import java.io.IOException;
-import java.io.OutputStream;
-import java.io.PrintStream;
-
-import static org.fusesource.jansi.internal.CLibrary.STDOUT_FILENO;
-import static org.fusesource.jansi.internal.CLibrary.isatty;
-import static org.fusesource.jansi.internal.Kernel32.GetConsoleMode;
-import static org.fusesource.jansi.internal.Kernel32.GetStdHandle;
-import static org.fusesource.jansi.internal.Kernel32.INVALID_HANDLE_VALUE;
-import static org.fusesource.jansi.internal.Kernel32.STD_OUTPUT_HANDLE;
-import static org.fusesource.jansi.internal.Kernel32.SetConsoleMode;
-
-/**
- * @see <a href="https://github.com/gradle/gradle/issues/882">Original issue (gradle/gradle#882)</a>
- * @see <a href="https://github.com/fusesource/jansi/issues/69">Issue in 3rd party library (fusesource/jansi#69)</a>
- */
-final class AnsiConsoleUtil {
-    private static final int ENABLE_VIRTUAL_TERMINAL_PROCESSING = 0x0004;
-    private static final int DISABLE_NEWLINE_AUTO_RETURN = 0x0008;
-
-    private static final boolean IS_XTERM = environmentVariableStartsWith("TERM", "xterm");
-    private static final boolean IS_MINGW_XTERM = IS_XTERM && environmentVariableStartsWith("MSYSTEM", "MINGW");
-
-    private static boolean environmentVariableStartsWith(String name, String pattern) {
-        return System.getenv(name) != null && System.getenv(name).startsWith(pattern);
-    }
-
-    private AnsiConsoleUtil() {
-    }
-
-    /**
-     * @see <a href="https://github.com/fusesource/jansi/blob/eeda18cb05122abe48b284dca969e2c060a0c009/jansi/src/main/java/org/fusesource/jansi/AnsiConsole.java#L48-L54">Method copied over from AnsiConsole.wrapOutputStream</a>
-     */
-    static OutputStream wrapOutputStream(final OutputStream stream) {
-        try {
-            return wrapOutputStream(stream, STDOUT_FILENO);
```

Comment:
This `STDOUT_FILENO` is weird as we called this function with both stdout and stderr streams (from how variables are called along the call tree). What are all the side-effects from hardcoding to stdout here?